added use-case for on session payments via PaymentRequest (#48)
MewsSystems/gitbook-distributor-guide: distributor-api-v1/operations.md

| `Client` | string | required | Identification of the client as described in [Authorization](https://mews-systems.gitbook.io/distributor-guide/distributor-api-v1/authorization). |
| `HotelId` | string | required | Unique identifier of the hotel. |
| `ReservationGroupId` | string | required | Unique identifier of the reservation group. |
| `Extent` | Object of [Extent](operations.md#reservation-group-get-extent) | optional | Object specifying whether to return related data |

### Extent <a id="reservation-group-get-extent"></a>
|  | Property | Type | Description |
| :--- | :--- | :--- | :--- |

> | `PaymentRequests` | boolean | optional | Whether the response should contain PaymentRequests related to the reservation group |

Reviewer: ```suggestion
| `PaymentRequests` | boolean | optional | Whether the response should contain PaymentRequests related to the reservation group. |
```

> | `Payments` | boolean | optional | Whether the response should contain Payment attempts related to the PaymentRequest for reservation group |

Reviewer: ```suggestion
| `Payments` | boolean | optional | Whether the response should contain Payment attempts related to the PaymentRequest for reservation group. |
```

### Response <a id="response-7"></a>

```json
{
    "Id": "f6fa7e62-eb22-4176-bc49-e521d0524dee",
    "CustomerId": "7ac6ca0d-7c08-4ab1-8da8-9b44979d8855",
    "Reservations": [],
    "TotalAmount": { },
    "PaymentRequestId": "2e3a700a-7b10-4e61-8e9f-acfa00ee00df",
    "PaymentRequests": [],
    "Payments": []
}
```

> This part provide documentation only for the related data in `PaymentRequests` & `Payments` rest of the response is the

Author: PaymentRequestId description was missing. I will add it to the `Create Reservation Group` (in table bellow [this example](https://github.com/MewsSystems/gitbook-distributor-guide/pull/48/files#diff-e0d8177d43ad27101a34daf5ee1404fa5b293c8884f331abcbe852ac6671a819R1086))

Mainly I do not want to describe the whole response here again. Basically it will be the same, but creation of `ReservationGroup` does not allow the extent so I would like to show and document the extra fields provided by the Extent

Let's discuss this in person, you have way more experiance with docs from Connector, so I am open to any suggestion 👍

same as in [Create Reservation Group](operations.md#create-reservation-group).

|  | Property | Type | Description |
| :--- | :--- | :--- | :--- |

> | `PaymentRequests` | Array of [PaymentRequests](operations.md#payment-request) | optional | Payment requests related to the reservation group. |
> | `Payments` | Array of [Payment](operations.md#payments) | optional | Payment attempts related to the Payment requests. |

Reviewer: Lower case and used singular so it matches other parts of docs, fixed anchor link (current doesn't exist) 
```suggestion
| `PaymentRequests` | array of [PaymentRequest](operations.md#payment-request) | optional | Payment requests related to the reservation group. |
| `Payments` | array of [Payment](operations.md#payment) | optional | Payment attempts related to the Payment requests. |
```

### PaymentRequest <a id="payment-request"></a>

|  | Property | Type | Description |
| :--- | :--- | :--- | :--- |

> | `Id` | string | required | Identifier of the payment request. |

Reviewer: So it matches naming in other parts of docs.
```suggestion
| `Id` | string | required | Unique identifier of the payment request. |
```

> | `ReservationGroupId` | string | required | Id of the related reservation group. |

Reviewer: So it matches naming in other parts of docs.

```suggestion
| `ReservationGroupId` | string | required | Unique identifier of the related reservation group. |
```

| `State` | string [PaymentRequestState](operations.md#payment-request-state) | required | State of the payment request. |

#### PaymentRequestState <a id="payment-request-state"></a>

- `Pending` - Non-finite state. Awaiting a next action.
- `Completed` - Finite state. Payment request that has been covered by payment.
- `Canceled` - Finite state. Payment request has been manually canceled by the creator (enterprise).

> - `Expired` - Finite state. Payment request has not been completed withing its expiration time.

Reviewer: ```suggestion
- `Expired` - Finite state. Payment request has not been completed within its expiration time.
```

Reviewer: @koprivajakub Just this one typo

### Payment <a id="payment"></a>

|  | Property | Type | Description |
| :--- | :--- | :--- | :--- |
| `Id` | string | required | Identifier of the payment. |
| `Amount` | [SingleCurrencyAmount](operations.md#single-currency-amount) | required | Amount in a currency which was used to create PaymentRequest - usually default currency of the enterprise. |
| `ChargeAmount` | [SingleCurrencyAmount](operations.md#single-currency-amount) | required | Amount in currency which was used for the payment during the charge. i.e. currency that will be visible on the user bank statement for the payment. |
| `CreatedUtc` | string | required | Date and time of the payment creation in UTC timezone in ISO 8601 format. |

> | `EnterpriseId` | string | required | Total amount of the reservation. |

Reviewer 2: EnterpriseId is amount of reservations?

| `PaymentRequestId` | string | required | Identifier of the payment request. |
| `State` | string [PaymentState](operations.md#payment-state) | required | State of the payment attempt. |

#### SingleCurrencyAmount <a id="single-currency-amount"></a>

> An object where name corresponds to ISO code and value represents a structure that holds gross price, net price and tax values.

Reviewer 2: What does mean that the name corresponds to iso code? what name?

Author: Copy paste error, sorry and thx

> An object where name corresponds to ISO code and value represents a structure that holds gross price, net price and tax values.

Reviewer: We already describe something similar in `Amount` https://mews-systems.gitbook.io/distributor-guide/distributor-api-v1/operations#amount . Maybe that section should be probably rewritten so that it says that `Amount` is an object where keys are currency codes and values are `SingleCurrencyAmounts`. And move the description of individual properties (GrossValue, NetValue, TaxValues) from there here.

- `Canceled` - Finite state. Payment request has been manually canceled by the creator (enterprise).
- `Expired` - Finite state. Payment request has not been completed within its expiration time.

### Payment <a id="payment"></a>

|  | Property | Type | Description |
| :--- | :--- | :--- | :--- |
| `Id` | string | required | Unique identifier of the payment. |
| `Amount` | [SingleCurrencyAmount](operations.md#single-currency-amount) | required | Amount in a currency which was used to create PaymentRequest - usually default currency of the enterprise. |
| `ChargeAmount` | [SingleCurrencyAmount](operations.md#single-currency-amount) | required | Amount in currency which was used for the payment during the charge. i.e. currency that will be visible on the user bank statement for the payment. |
| `CreatedUtc` | string | required | Date and time of the payment creation in UTC timezone in ISO 8601 format. |
| `EnterpriseId` | string | required | Identifier of the enterprise receiving the payment. |
| `PaymentRequestId` | string | required | Identifier of the payment request. |
| `State` | string [PaymentState](operations.md#payment-state) | required | State of the payment attempt. |

> #### SingleCurrencyAmount <a id="single-currency-amount"></a>

Author: True and done

```json
{
    "Currency": "EUR",
    "GrossValue": 115.00,
    "NetValue": 115.00,
    "TaxValues": []
}
```

| Property | Type | Description |
| :--- | :--- | :--- |
| `Currency` | string | ISO 4217 code of the currency. |

> | `GrossValue` | number | Value of the payment amount. |

Author: Well there is nothing like `Gross` and `Net` in the payment. The values will be identical. But I can change it, let's discuss this on call.

> | `NetValue` | number | Value of the payment amount. |

Author: see the previous comment
